### DISCUSSION TITLE
Feature Req: Loading env file from within test

### DISCUSSION THREAD

**Discussion by kjs3** [jdx/mise#3796] [Ideas]
This doesn't work:

```sh
#!/usr/bin/env sh

# MISE env={_.file = ".env"}

echo "NODE ENV: $NODE_ENV"
```

But it would be fantastic if it did! 🙂

Or some other way to dynamically load vars so my various .env files can be shared with tasks.

Importantly, I need different tasks to load different .env files within a monorepo.

**Comment by syhol** [comment]
Until this is implemented, it can be emulated like this:
```shell
#!/usr/bin/env sh
#MISE env={SOURCE_FILE = "{{config_root}}/.env"}
source $SOURCE_FILE
echo "NODE ENV: $NODE_ENV"
```

**Reply by kjs3** [reply]
Thank you!

**Comment by Marukome0743** [comment]
This is now supported by [PR #5638](https://github.com/jdx/mise/pull/5638), with file-task header parsing improved further in [PR #8574](https://github.com/jdx/mise/pull/8574).

File tasks can load an environment file using a task-specific `_.file` directive:

```sh
#!/usr/bin/env sh
#MISE env._.file = ".env"

echo "NODE ENV: $NODE_ENV"
```

The environment file is loaded for that task, so different file tasks in a monorepo can point to different `.env` files without adding them to the project-wide mise environment. The manual `source` workaround is therefore no longer necessary.

One syntax detail: the task header marker must be written as `#MISE` without a space between `#` and `MISE`. PR #8574 added support for spaces around the `=` itself, as shown above.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
